.github/workflows/unitTest.yml

```
@@ -25,5 +25,19 @@ jobs:
+        set -o pipefail && xcodebuild test \
+          -scheme three-days-UnitTest \
+          -destination 'platform=iOS Simulator,name=iPhone 15,OS=latest' \
+          | tee result.log
+        if grep -q "** TEST FAILED **" result.log; then
+          exit 1
+        fi
+      continue-on-error: false
+
+    - name: Failure notification to Discord
+      if: failure()
+      run: |
```

actionlint

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)